bigtable/admin/column_family.h

```diff
@@ -0,0 +1,150 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_ADMIN_COLUMN_FAMILY_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_ADMIN_COLUMN_FAMILY_H_
+
+#include "bigtable/client/version.h"
+
+#include <chrono>
+#include <memory>
+
+#include <google/bigtable/admin/v2/table.pb.h>
+
+#include "bigtable/client/detail/conjunction.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * Implement a thin wrapper around google::bigtable::admin::v2::GcRule.
+ *
+ * Provides functions to create GcRules in a convenient form.
+ */
+class GcRule {
+ public:
+  /// Create a garbage collection rule that keeps the last @p n versions.
+  static GcRule MaxNumVersions(std::int32_t n) {
+    GcRule tmp;
+    tmp.gc_rule_.set_max_num_versions(n);
+    return tmp;
+  }
+
+  /**
+   * Return a garbage collection rule that deletes cells in a column older than
+   * the given duration.
+   *
+   * The function accepts any instantiation of std::chrono::duration<> for the
+   * @p duration parameter.
+   *
+   * @tparam Rep a placeholder to match the Rep tparam for @p duration type, the
```

Comment:
I find all of this explanation confusing but it's actually not hard to use. Maybe lead with an example for these?